File: backend/src/main/java/nextpos/app/nextpos/service/interf/PaymentService.java
package nextpos.app.nextpos.service.interf;

import nextpos.app.nextpos.model.dto.request.CreatePaymentRequest;
import nextpos.app.nextpos.model.dto.request.UpdatePaymentRequest;
import nextpos.app.nextpos.model.dto.response.PaymentResponse;
import nextpos.app.nextpos.model.enums.PaymentSourceType;

import java.util.List;

public interface PaymentService {

    PaymentResponse createPayment(CreatePaymentRequest request);

    PaymentResponse updatePayment(Long id, UpdatePaymentRequest request);

    void deletePayment(Long id);

    PaymentResponse getPayment(Long id);

    List<PaymentResponse> getPaymentsByReference(PaymentSourceType referenceType, Long referenceId);
    
    PaymentResponse processPayment(Long saleId, CreatePaymentRequest request);
}

File: frontend/src/services/paymentService.ts
import { PaymentSourceType } from "@/enums/paymentSourceType";
import api from "./api";
import type {
  CreatePaymentRequest,
  UpdatePaymentRequest,
  PaymentResponse,
} from "@/types/Payment";

export interface ApiResponse<T> {
  success: boolean;
  message: string | null;
  data: T;
}

export const createPayment = (data: CreatePaymentRequest) =>
  api.post<ApiResponse<PaymentResponse>>("/payments", data);

export const updatePayment = (id: number, data: UpdatePaymentRequest) =>
  api.put<ApiResponse<PaymentResponse>>(`/payments/${id}`, data);

export const deletePayment = (id: number) =>
  api.delete<void>(`/payments/${id}`);

export const getPaymentById = (id: number) =>
  api.get<ApiResponse<PaymentResponse>>(`/payments/${id}`);

export const getPaymentsByReference = (
  type: PaymentSourceType,
  referenceId: number,
) =>
  api.get<ApiResponse<PaymentResponse[]>>("/payments/reference", {
    params: { type, referenceId },
  });

export const getPaymentStatus = (id: number) =>
  api.get<ApiResponse<{ status: string }>>(`/payments/${id}/status`);
